Update billing documentation (#511)
Shopify/shopify-node-api: docs/usage/billing.md

| `replacementBehavior` | `BillingReplacementBehavior` |    No     |       -       | `BillingReplacementBehavior` value, see [the reference](https://shopify.dev/api/admin-graphql/2022-07/mutations/appSubscriptionCreate) for more information. Not available for `OneTime` plans |

## Checking for payment

Once the app has an access token, you can call `shopify.billing.check` to look for an existing payment.

Here's a typical example of how apps might do that:

```ts
// This can happen at any point after the merchant goes through the OAuth process
function billingMiddleware(req, res, next) {
  const session = shopify.session.getCurrent({
    isOnline: true,
    rawRequest: req,
    rawResponse: res,
  });

  const hasPayment = await shopify.billing.check({
    session,
    plans: ['My billing plan'],
    isTest: true,
  });

  if (hasPayment) {
    next();
  } else {
    // Either request payment now or redirect to plan selection page
  }
}

app.use('/requires-payment/*', billingMiddleware);
```

> **Note**: this method will always query Shopify's API because merchants can cancel subscriptions at any point. Depending on the number of requests your app handles, you might want to cache a merchant's payment status, but you should make sure to periodically call this method to ensure you're blocking unpaid access.

Reviewer: ```suggestion
**Note**: the `check` method will always query Shopify's API because merchants can cancel subscriptions at any point. Depending on the number of requests your app handles, you might want to cache a merchant's payment status, but you should periodically call this method to ensure you're blocking unpaid access.
```

The `check` method accepts the following parameters:

| Parameter | Type                 | Required? | Default Value | Notes                                   |
| --------- | -------------------- | :-------: | :-----------: | --------------------------------------- |
| `session` | `Session`            |    Yes    |       -       | The `Session` for the current request   |
| `plans`   | `string \| string[]` |    Yes    |       -       | Which plans to look for                 |
| `isTest`  | `boolean`            |    No     |    `true`     | Whether to look for test purchases only |

It returns `true` if there is a payment for any of the given plans, and `false` otherwise.

## Requesting payment

When you want create a new charge for the merchant, you can call `shopify.billing.request` to instruct Shopify to create a charge.
Here are some typical examples of how apps might do that:

<div>Example single-plan setup - charge after OAuth completes

```ts
app.get('/auth/callback', async () => {
  const callback = await shopify.auth.callback({
    isOnline: false,
    rawRequest: req,
    rawResponse: res,
  });

>   // Check if we require payment

Reviewer: ```suggestion
  // Check if we require payment ... see example above
```

# Billing for your app

Some partners might wish to charge merchants for using their app.
The Admin API provides endpoints that enable apps to trigger purchases in the Shopify platform, so that merchants can pay for apps as part of their Shopify payments.

This library provides support for billing apps by:

1. Checking if the store has already paid for the app

> 1. Triggering a charge for the merchant if not

Reviewer: ```suggestion
1. Triggering a charge for the merchant if the store has not paid
```

```ts
app.post('/api/select-plan', async (req, res) => {
  const session = shopify.session.getCurrent({
    isOnline: true,
    rawRequest: req,
    rawResponse: res,
  });

  const confirmationUrl = await shopify.billing.request({
    session,
    // Receive the selected plan from the frontend
    plan: req.body.selectedPlan,
    isTest: true,
  });

  res.redirect(confirmationUrl);
});
```

The `request` method accepts the following parameters:

| Parameter | Type      | Required? | Default Value | Notes                                                |
| --------- | --------- | :-------: | :-----------: | ---------------------------------------------------- |
| `session` | `Session` |    Yes    |       -       | The `Session` for the current request                |
| `plan`    | `string`  |    Yes    |       -       | Which plan to create a charge for                    |
| `isTest`  | `boolean` |    No     |    `true`     | If `true`, Shopify will not charge for this purchase |

It will return a `string` containing a URL - we don't redirect right away to make it possible for apps to run their own code after it creates the payment request.

> The app must redirect the merchant to this URL so that they can confirm the charge before Shopify applies it.

Reviewer: ```suggestion
The app **must** redirect the merchant to this URL so that they can confirm the charge before Shopify applies it.
```
